api/securitytest/spotbugs.go

```diff
@@ -0,0 +1,146 @@
+type Error struct {
+	XMLName        xml.Name `xml:"Errors"`
+	Errors         string   `xml:"errors,attr"`
+	MissingClasses string   `xml:"missingClasses,attr"`
+}
+
+// SourceLine is the struct that holds details about issue location
+type SourceLine struct {
+	XMLName       xml.Name `xml:"SourceLine"`
+	ClassName     string   `xml:"classname,attr"`
+	Start         string   `xml:"start,attr"`
+	End           string   `xml:"end,attr"`
+	StartByteCode string   `xml:"startBytecode,attr"`
+	EndByteCode   string   `xml:"endBytecode,attr"`
+	SourceFile    string   `xml:"sourcefile,attr"`
+	SourcePath    string   `xml:"sourcepath,attr"`
+}
+
+func analyzeSpotBugs(spotbugsScan *SecTestScanInfo) error {
+
+	spotBugsOutput := SpotBugsOutput{}
+	spotbugsScan.FinalOutput = spotBugsOutput
+
+	// nil cOutput states that no Issues were found.
+	if spotbugsScan.Container.COutput == "" {
+		spotbugsScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// Unmarshall rawOutput into finalOutput, that is a SpotBugsOutput struct.
+	spotBugsOutput, err := parseXMLtoJSON([]byte(spotbugsScan.Container.COutput))
+	if err != nil {
+		log.Error("analyzeSpotBugs", "SPOTBUGS", 1039, spotbugsScan.Container.COutput, err)
+		spotbugsScan.ErrorFound = err
+		spotbugsScan.prepareContainerAfterScan()
+		return err
+	}
+
+	spotbugsScan.FinalOutput = spotBugsOutput
+
+	// check results and prepare all vulnerabilities found
+	spotbugsScan.prepareSpotBugsVulns()
+	spotbugsScan.prepareContainerAfterScan()
+	return nil
+}
+func parseXMLtoJSON(byteValue []byte) (SpotBugsOutput, error) {
+	var bugs SpotBugsOutput
+	if err := xml.Unmarshal(byteValue, &bugs); err != nil {
+		return bugs, err
+	}
+	numOfErrors, err := strconv.Atoi(bugs.Errors.Errors)
+	if err != nil {
+		return bugs, err
+	}
+	numOfMissingClasses, err := strconv.Atoi(bugs.Errors.MissingClasses)
+	if err != nil {
+		return bugs, err
+	}
+	if (len(bugs.SpotBugsIssue) == 0) && (numOfErrors > 0 || numOfMissingClasses > 0) {
+		return bugs, fmt.Errorf("spotbugs has risen errors because of [%d] missing classes and [%d] errors while analysing", numOfMissingClasses, numOfErrors)
+	}
+	return bugs, nil
+}
+
+func (spotbugsScan *SecTestScanInfo) prepareSpotBugsVulns() {
+	var spotbugsOutput SpotBugsOutput
+
+	huskyCIspotbugsResults := types.HuskyCISecurityTestOutput{}
+	spotbugsOutput = spotbugsScan.FinalOutput.(SpotBugsOutput)
+
+	for i := 0; i < len(spotbugsOutput.SpotBugsIssue); i++ {
+		for j := 0; j < len(spotbugsOutput.SpotBugsIssue[i].SourceLine); j++ {
+			spotbugsVuln := types.HuskyCIVulnerability{}
+			spotbugsVuln.Language = "Java"
+			spotbugsVuln.SecurityTool = "SpotBugs"
+			spotbugsVuln.Type = spotbugsOutput.SpotBugsIssue[i].Abbreviation
+			spotbugsVuln.Details = spotbugsOutput.SpotBugsIssue[i].Type
+			spotbugsVuln.File = spotbugsOutput.SpotBugsIssue[i].SourceLine[j].SourcePath
+			spotbugsVuln.Code = spotbugsOutput.SpotBugsIssue[i].SourceLine[j].SourcePath
```

Comment:
This `Code` is duplicated with the `File` right above it. It would be great if we have something like this: 
```
startLine := spotbugsOutput.SpotBugsIssue[i].SourceLine[j].Start
endLine := spotbugsOutput.SpotBugsIssue[i].SourceLine[j].End 
spotbugsVuln.Code = fmt.Sprintf("Code beetween Line %s and Line %s.",startLine, endLine)
spotbugsVuln.Line = startLine
```

What do you think? 🙃